### PR TITLE
119 get donor campaigns route

### PR DESCRIPTION
**Pull request by farah2003**
relates #123 
colse #123

### CODE REVIEW

File: server/routes/adminDonor.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { Router } from 'express';
-import { deleteDonorById } from '../controllers';
+import { deleteDonorById, getDonorCampaigns } from '../controllers';
 import { authUser, authAdmin } from '../middlewares';
 
 const adminDonorRouter = Router();
 
 adminDonorRouter.route('/donor/:id').delete(authUser, authAdmin, deleteDonorById);
-
+adminDonorRouter.get('/donor/campagins/:id', authUser, authAdmin, getDonorCampaigns);
```
Comment (Mohammed-Hamada) [inline server/routes/adminDonor.ts:8]: what if you use ```/donor/:id/campagins``` instead ```/donor/campagins/:id```, isn't that more clearly and readable?

**Review by Mohammed-Hamada — CHANGES_REQUESTED** [review]

**Review by DeenaAlghazali — APPROVED** [review]

File: server/controllers/admin/donors/getDonorsCampaigns.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      include: {
+        model: Campaign,
+        attributes: ['title'],
+        through: {
+          attributes: [],
+        },
+      },
+      attributes: [],
+    });
+    if (!campaigns) throw new CustomError('There is no donor', 400);
```
Comment (HaniOlwan) [inline server/controllers/admin/donors/getDonorsCampaigns.ts:18]: if there's no campaigns for this donor return empty array

File: server/routes/adminDonor.ts (hunk 1 above)
Comment (HaniOlwan) [reply server/routes/adminDonor.ts:8]: yes i agree

File: server/controllers/admin/donors/getDonorsCampaigns.ts (hunk 2 above)
Comment (HaniOlwan) [reply server/controllers/admin/donors/getDonorsCampaigns.ts:18]: also return the id for the campaign in case we want to visit that campaign

**Review by HaniOlwan — APPROVED** [review]

File: server/controllers/admin/donors/getDonorsCampaigns.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    const { id } = await paramsSchema.validateAsync(req.params);
+    const campaigns = await Donor.findByPk(id, {
+      include: {
+        model: Campaign,
+        attributes: ['title'],
+        through: {
+          attributes: [],
+        },
+      },
+      attributes: [],
```
Comment (Raghad-Mezied) [inline server/controllers/admin/donors/getDonorsCampaigns.ts:16]: Why do you return nothing from the donor?

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by Mohammed-Hamada — APPROVED** [review]

**Pull request merged**